markdown/frameworks/framework_einstieg.md

```diff
@@ -0,0 +1,146 @@
+---
+type: lecture-cg
+title: "Frameworks: Einstieg"
+menuTitle: "Frameworks"
+author: "André Matutat (FH Bielefeld)"
+weight: 1
+readings:
+  - key: ""
+tldr: |
+  Frameworks sind ein Softwaregerüst, um wiederkehrende Softwarestrukturen schnell und einfach um die eigene Funktionalität zu erweitern.
+  Im Gegensatz zu Libraries stellen sie keine einzelnen Funktionen bereits, die im eigene Code aufgerufen werden, sondern geben Strukturen vor, in denen der eigenen Code eingearbeitet werden muss.
+outcomes:
+  - k1: "Begriff Framework kennen gelernt"
+  - k2: "Unterschied zwischen Frameworks und Libraries verstehen"
+quizzes:
+  - link: "XYZ"
+    name: "Quiz XXX (ILIAS)"
+youtube:
+  - link: ""
+    name: "VL "
+  - link: ""
+    name: "Demo "
+  - link: ""
+    name: "Demo "
+fhmedia:
+  - link: ""
+    name: "VL "
+sketch: true
+---
+
+
+## Motivation
```

Comment:
Der Text ist super fürs Skript (also bitte drin lassen und in ein `::: notes ... :::` setzen), aber für die Folie taugt das nicht. Ich würde gern mit einem Problem o.ä. ins Haus fallen. Am besten so, dass es auch gut auf eine Folie passt (Bild oder Code-Schnipsel oder 3..4 Stichpunkte)

Reply:
guter Einwand.
Am liebsten hätte ich irgendwas ekliges, wenn man es per Hand macht. 
Eigener WebServer oder sowas.

Reply:
ja, wegen mir. aber es muss auf ne folie passen.

Reply:
das wird das problem sein

Reply:
:)

Reply:
Ich komm hier irgendwie nicht so richtig weiter.

Eigentlich hätte ich gerne Code (habe an die [draw](https://github.com/libgdx/libgdx/blob/master/gdx/src/com/badlogic/gdx/graphics/g2d/SpriteBatch.java) Methode von libGDX gedacht). Das ist viel (und ekliger) Code, den man für jede Anwendung 1zu1 wieder neu schreiben müssten. Aus dem Kontext gerissen, klingt das aber ehr nach einem Beispiel warum Libraries nützlich sind.

Alternativ hätte ich eine "Checkliste" für Videogames per Hand bauen. Also was muss man alles erstellen/beachten bevor man überhaupt das eigentliche Spiel programmieren kann.

- [ ] GameLoop anlegen
- [ ] Schnittstellen für Benutzereingaben erstellen
- [ ] Koordinatensystem implementieren
- [ ] Kollisionsabfrage 
- [ ] Anwendungsfenster erstellen/zeichnen
- [ ] Funktionen um Objekte im Fenster zu zeichnen
- [ ] Zeichnungen regelmäßig aktualisieren
- [ ] Kameras implementieren
- [ ] ggf. Netzwerkfunktionen

Reply:
hmmm. schreib doch mal den code auf und verdichte ihn. vielleicht passt das ja noch? 

ansonsten könnte die checkliste auch ne gute idee sein. leider kommt markdown hier an seine grenzen: man könnte in einer zweiten folie (in powerpoint wäre das einfach eine animation) die checkliste bunt markieren, was quasi durch das framework abgedeckt ist. 

vielleicht als bild (checkliste als ein bild, und dann nochmal das selbe bild mit bunten rahmen oder so)?

Reply:
das mit den zwei checklisten folien klingt gut.
ich verfeiner die liste mal und bastel was zusammen

Reply:
hab mal was gebastelt.

Reply:
ja, könnte gehen. letztlich ist es ja die ton-spur, die hier den sinn macht. die folie ist ja nur "optik" ;)